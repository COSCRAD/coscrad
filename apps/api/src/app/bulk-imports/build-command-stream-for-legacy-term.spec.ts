import {
    LegacyTerm,
    buildCommandStreamForLegacyTerm,
} from './build-command-stream-for-legacy-term';

const dummyOldId = '45678';

const termText = 'I am running (lang)';

const termEnglishText = 'I am running (engl)';

const contributorWithPromptTerms = '1';

const _contributorWithTranslationTerms = '2';

const audioFilename = 'scrabble.mp3';

const sourceId = '2';

describe(`buildCommandStreamForLegacyTerm`, () => {
    describe(`when the term has text in both languges`, () => {
        describe(`it should emit the correct commands`, () => {
            const legacyTerm: LegacyTerm = {
                _key: dummyOldId,
                _id: `terms/${dummyOldId}`,
                _rev: '_ds6eKEK---',
                published: true,
                term: termText,
                termEnglish: termEnglishText,
                contributorId: contributorWithPromptTerms,
                audioFilename,
                sourceId,
            };

            const result = buildCommandStreamForLegacyTerm(legacyTerm);

            it(`should return the expected command stream`, () => {
                /**
                 * Visual inspection works well for this task.
                 */
                expect(result).toMatchSnapshot();
            });
        });
    });
});
