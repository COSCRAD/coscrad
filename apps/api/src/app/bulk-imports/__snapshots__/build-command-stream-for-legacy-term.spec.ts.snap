// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildCommandStreamForLegacyTerm when the term has text in both languges it should emit the correct commands should return the expected command stream 1`] = `
[
  {
    "payload": {
      "aggregateCompositeIdentifier": {
        "id": "45678",
        "type": "term",
      },
      "contributorId": "1",
      "languageCode": "clc",
      "rawData": {
        "_id": "terms/45678",
        "_key": "45678",
        "_rev": "_ds6eKEK---",
        "audioFilename": "scrabble.mp3",
        "contributorId": "1",
        "published": true,
        "sourceId": "2",
        "term": "I am running (lang)",
        "termEnglish": "I am running (engl)",
      },
      "text": "I am running (lang)",
    },
    "type": "CREATE_TERM",
  },
  {
    "payload": {
      "aggregateCompositeIdentifier": {
        "id": "45678",
        "type": "term",
      },
      "languageCode": "en",
      "translation": "I am running (engl)",
    },
    "type": "TRANSLATE_TERM",
  },
  {
    "payload": {
      "aggregateCompositeIdentifier": {
        "id": "45678",
        "type": "term",
      },
    },
    "type": "PUBLISH_RESOURCE",
  },
]
`;
